File: Programmers/Lv.2.Kotlin/행렬의곱셈.kt
class Solution {
    fun solution(arr1: Array<IntArray>, arr2: Array<IntArray>): Array<IntArray> {
        var answer = Array<IntArray>(arr1.size) { IntArray(arr2.first().size){ 0 } }
        
        for (row in arr1.indices) {
            for (col2 in arr2.first().indices) {
                for (col in arr1[row].indices) {
                    answer[row][col2] += arr1[row][col] * arr2[col][col2]
                }
            }
        }
        
        return answer
    }
}

File: Programmers/Lv.2/행렬의곱셈.py
def solution(arr1, arr2):
    answer = [[0 for _ in range(len(arr2[0]))] for _ in range(len(arr1))]
    for i in range(len(arr1)):
        for j in range(len(arr2[0])):
            for k in range(len(arr1[0])):
                answer[i][j] += arr1[i][k] * arr2[k][j]
                
    return answer

# def solution(A, B):
#     return [[sum(a * b for a, b in zip(A_row, B_col)) for B_col in zip(* B)] for A_row in A]